lib/ansible/modules/cloud/scaleway/scaleway_organization_facts.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2018, Yanis Guenane <yanis+ansible@guenane.org>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: scaleway_organization_facts
+short_description: Gather facts about the Scaleway organizations available.
+description:
+  - Gather facts about the Scaleway organizations available.
+version_added: "2.7"
+author:
+  - "Yanis Guenane (@Spredzy)"
+  - "Remy Leone (@sieben)"
+extends_documentation_fragment: scaleway
+'''
+
+EXAMPLES = r'''
+- name: Gather Scaleway organizations facts
+  scaleway_organization_facts:
+'''
+
+RETURN = r'''
+---
+scaleway_organization_facts:
+  description: Response from Scaleway API
+  returned: success
+  type: complex
+  contains:
+    "scaleway_organization_facts": [
+        {
+            "address_city_name": "Paris",
+            "address_country_code": "FR",
+            "address_line1": "42 Rue de l'univers",
+            "address_line2": null,
+            "address_postal_code": "75042",
+            "address_subdivision_code": "FR-75",
+            "creation_date": "2018-08-06T13:43:28.508575+00:00",
+            "currency": "EUR",
+            "customer_class": "individual",
+            "id": "3f709602-5e6c-4619-b80c-e8432ferewtr",
+            "locale": "fr_FR",
+            "modification_date": "2018-08-06T14:56:41.401685+00:00",
+            "name": "James Bond",
+            "support_id": "694324",
+            "support_level": "basic",
+            "support_pin": "9324",
+            "users": [],
+            "vat_number": null,
+            "warnings": []
+        }
+    ]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.scaleway import (
+    Scaleway, ScalewayException, scaleway_argument_spec
+)
+
+
+class ScalewayOrganizationFacts(Scaleway):
+
+    def __init__(self, module):
+        super(ScalewayOrganizationFacts, self).__init__(module)
+        self.name = 'organizations'
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=scaleway_argument_spec(),
+        supports_check_mode=True,
+    )
+
+    module.params['api_url'] = 'https://account.scaleway.com'
```

Comment:
Is it normal this is hardcoded here and not on others ?

Reply:
Yes because this module use this specific API

Reply:
@mscherer  Most modules uses https://api.scaleway.com (This is the default for `api_url`).

Two endpoints use actually the https://account.scaleway.com - organization and user - (this is the way the Scaleway API is designed https://developer.scaleway.com/). To avoid user to have to know those details, it is hardcoded here.

Reply:
But so, the idea was that someone could override the api_url for some reasons (I assumed future proofing). So that mean that for this module, the documentation saying that api_url go to api.scaleway.com will be wrong, and that using the api_url argument will not work.  I am not sure why we have a argument api_url in the first place, but if we do have a reason, i think the reason to change would also apply here, no ?

Reply:
@mscherer in my understanding not really. There is an API url for the other resources because they can hit either `https://cp-par1.scaleway.com` (Paris) or `https://cp-ams1.scaleway.com` (Amsterdam) or `https://api.scaleway.com` (I think the one attached to your profile). Those are resources related urls. For the account related url there is just one url available `https://account.scaleway.com` (For `/user` and `/organization`). This is done here exactly to abstract those many urls from the user, but for account related URL there is no need to be able to specify the url throught `api_url`.

Reply:
So, any way to not expose that as a argument ? Cause ansible-doc do show it with wrong value.

Reply:
@mscherer done. Unfortunately, ansible-doc still shows the wrong information, but the proper information are available in the argument spec. Not sure if its a bug or not

Reply:
@mscherer please discard my last comment, it was too late at night ;) Doc is working now a expected and show the proper api URL based on the context.